### PR TITLE
chore: feature flag resiliency

### PR DESCRIPTION
**Pull request by EDsCODE**
**What does this PR do?**

- resiliency points for https://github.com/PostHog/posthog/issues/13627

**Where should the reviewer start?**

**How should this be manually tested?**

**Any background context you want to provide?**

**What are the relevant tickets?**

**Screenshots or screencasts (if UI/UX change)**

**Questions:**
- Does the docs need an update?
- Are there any security concerns?
- Do we need to update engineering / success?

### CODE REVIEW

**Comment by liyiy** [comment]
should we add tests for this? (asking for android too)

**Comment by EDsCODE** [comment]
Good point, the no-replacement on error yes I would say so. For timeouts idts

File: PostHog/Internal/PHGHTTPClient.m [hunk 1]
```diff
@@ -140,9 +142,18 @@ - (NSURLSessionDataTask *)sharedSessionUpload:(NSDictionary *)payload host:(NSUR
                 PHGLog(@"Error uploading request %@.", error);
                 failure(error);
             } else {
-                NSDictionary *json  = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
-                NSLog(@"%@",json);
-                success(json);
+                NSInteger code = ((NSHTTPURLResponse *)response).statusCode;
+                if (code < 300) {
```
Comment (neilkakkar) [inline PostHog/Internal/PHGHTTPClient.m:146]: can this be anything other than 200?

Comment (EDsCODE) [reply PostHog/Internal/PHGHTTPClient.m:146]: Do you mean based on what we send?

**Pull request merged**